corehq/apps/campaign/models.py

```diff
@@ -95,3 +95,35 @@ class DashboardGauge(DashboardWidgetBase):
 
     # optional additional configuration set to customize gauge appearance
     configuration = JSONField(default=dict)
+
+
+class WidgetType:
+    from corehq.apps.campaign.forms import DashboardMapForm, DashboardReportForm
+
+    MAP = 'map'
+    REPORT = 'report'
+
+    FORM_CLASS = {
+        MAP: DashboardMapForm,
+        REPORT: DashboardReportForm,
+    }
+
+    MODEL_CLASS = {
+        MAP: DashboardMap,
+        REPORT: DashboardReport,
+    }
+
+    @classmethod
+    def get_form_class(cls, widget_type):
+        return cls.FORM_CLASS.get(widget_type)
+
+    @classmethod
+    def get_model_class(cls, widget_type):
+        return cls.MODEL_CLASS.get(widget_type)
+
+    @classmethod
+    def choices(cls):
+        return [
+            (cls.MAP, _('Map')),
+            (cls.REPORT, _('Report')),
+        ]
```

Comment:
TextChoices gives you your `choices()` method for free. You could do something similar to the following.

```suggestion
class WidgetType(models.TextChoices):
    MAP = 'map', _('Map')
    REPORT = 'report', _('Report')

    @classmethod
    def get_form_class(cls, widget_type):
        from corehq.apps.campaign.forms import (
            DashboardMapForm,
            DashboardReportForm,
        )

        form_classes = {
            cls.MAP: DashboardMapForm,
            cls.REPORT: DashboardReportForm,
        }
        return form_classes[widget_type]

    @classmethod
    def get_model_class(cls, widget_type):
        return cls.get_form_class(widget_type).Meta.model
```

Reply:
This is amazing.  Addressed in https://github.com/dimagi/commcare-hq/pull/35995/commits/7531f126383ec6a8361ed94532e84e92e9333cc6